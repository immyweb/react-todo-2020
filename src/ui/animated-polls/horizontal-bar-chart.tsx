import * as React from 'react';
import * as DOM from 'react-dom';

import { fonts } from '../../styles/fonts';

import { ResponsiveContainer, BarChart, Bar, Cell, XAxis, YAxis, LabelList } from 'recharts';

import { colours } from '../../styles/colours';

import { IVoteCount } from './types';
import { ITheme } from '../../types/theme';

interface IProps {
  data: IVoteCount[];
  theme: ITheme;
}

const barGap = 10;

export class ChartHorizontalBar extends React.Component<IProps, { width: number; height: number }> {
  private chartElement: HTMLElement;

  constructor(props: IProps) {
    super(props);
    this.state = {
      width: 0,
      height: 0,
    };
  }

  componentDidMount() {
    this.chartElement = DOM.findDOMNode(this) as HTMLElement;
    this.setDimensions();
    window.addEventListener('resize', this.setDimensions.bind(this));
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.setDimensions.bind(this));
  }

  setDimensions = () => {
    this.setState({
      width: this.chartElement.offsetWidth,
      height: this.chartElement.offsetHeight,
    });
  };

  renderPercentageLabel = (props: any) => {
    return this.renderText({
      ...props,
      x: this.state.width,
      value: `${props.value}%`,
      textAnchor: 'end',
    });
  };

  renderOptionLabel = (props: any) => {
    return this.renderText({
      ...props,
      x: this.getThumbSize() + 20,
      textAnchor: 'start',
    });
  };

  getThumbSize = () => {
    return this.state.height / this.props.data.length - barGap;
  };

  renderText = (props: any) => {
    const { x, y, height, value, textAnchor } = props;
    const fontSize = 16;
    return (
      <text
        x={x}
        y={y + (fontSize + height) / 2 - 5}
        fill={colours.$white}
        textAnchor={textAnchor}
        style={{ fontSize, fontFamily: fonts.$theSunRegular }}
        dominantBaseline="middle"
      >
        {value}
      </text>
    );
  };

  renderImageLabels = (props: any) => {
    const { x, y, value } = props;
    return (
      <image
        // height={this.getThumbSize()}
        width={35}
        xlinkHref={value}
        x={x + 8}
        y={y + 5}
      />
    );
  };

  render() {
    const { data, theme } = this.props;
    return (
      <ResponsiveContainer height={'100%'} width={'100%'}>
        <BarChart data={data} layout={'vertical'} barGap={barGap} margin={{ right: 49 }}>
          <XAxis domain={[0, 100]} type={'number'} axisLine={false} hide={true} />
          <YAxis dataKey={'player'} type={'category'} hide={true} />
          <Bar
            dataKey={'votePercentage'}
            barSize={this.getThumbSize()}
            background={{ fill: theme.secondary, fillOpacity: 1.0 }}
          >
            <LabelList
              dataKey={'votePercentage'}
              position={'right'}
              content={this.renderPercentageLabel}
            />
            <LabelList
              dataKey={'player'}
              position={'insideRight'}
              content={this.renderOptionLabel}
            />
            <LabelList
              dataKey={'iconImg'}
              position={'insideRight'}
              content={this.renderImageLabels}
            />
            {data.map((entry: IVoteCount, i: number) => {
              return <Cell key={i} fill={theme.primary} />;
            })}
          </Bar>
        </BarChart>
      </ResponsiveContainer>
    );
  }
}
