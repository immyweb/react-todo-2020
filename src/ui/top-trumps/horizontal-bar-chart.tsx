import * as React from 'react';
import * as DOM from 'react-dom';

import { fonts } from '../../styles/fonts';

import { ResponsiveContainer, BarChart, Bar, Cell, XAxis, YAxis, LabelList } from 'recharts';

import { colours } from '../../styles/colours';

import { IVoteCount } from './types';
import { ITheme } from '../../types/theme';

interface IProps {
  data: IVoteCount[];
  theme: ITheme;
}

const barGap = 10;

export class ChartHorizontalBar extends React.Component<IProps, { width: number; height: number }> {
  private chartElement: HTMLElement;

  constructor(props: IProps) {
    super(props);
    this.state = {
      width: 0,
      height: 0,
    };
  }

  componentDidMount() {
    this.chartElement = DOM.findDOMNode(this) as HTMLElement;
    this.setDimensions();
    window.addEventListener('resize', this.setDimensions.bind(this));
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.setDimensions.bind(this));
  }

  setDimensions = () => {
    this.setState({
      width: this.chartElement.offsetWidth,
      height: this.chartElement.offsetHeight,
    });
  };

  renderPercentageLabel = (props: any) => {
    return this.renderText({
      ...props,
      x: this.state.width - 15,
      value: `${props.value}%`,
      textAnchor: 'end',
    });
  };

  renderPlayerLabel = (props: any) => {
    return this.renderText({
      ...props,
      x: this.getThumbSize() + 20,
      textAnchor: 'start',
    });
  };

  getThumbSize = () => {
    return this.state.height / this.props.data.length - barGap;
  };

  renderText = (props: any) => {
    const { x, y, height, value, textAnchor } = props;
    const fontSize = 16;
    return (
      <text
        x={x}
        y={y + (fontSize + height) / 2 - 3}
        fill={colours.$white}
        textAnchor={textAnchor}
        style={{ fontSize, fontFamily: fonts.$theSunRegular }}
        dominantBaseline="middle"
      >
        {value}
      </text>
    );
  };

  renderImageLabels = (props: any) => {
    const { x, y, value } = props;
    return (
      <image
        height={this.getThumbSize()}
        width={this.getThumbSize()}
        xlinkHref={value}
        x={x - this.getThumbSize() - 2}
        y={y}
      />
    );
  };

  render() {
    const { data, theme } = this.props;
    return (
      <ResponsiveContainer height={'100%'} width={'100%'}>
        <BarChart
          data={data}
          layout={'vertical'}
          barGap={barGap}
          margin={{ left: this.getThumbSize() + 5 }}
        >
          <XAxis domain={[0, 100]} type={'number'} axisLine={false} hide={true} />
          <YAxis dataKey={'player'} type={'category'} hide={true} />
          <Bar
            dataKey={'votePercentage'}
            barSize={this.getThumbSize()}
            background={{ fill: colours.$midGrey, fillOpacity: 0.4 }}
          >
            <LabelList
              dataKey={'votePercentage'}
              position={'insideRight'}
              content={this.renderPercentageLabel}
            />
            <LabelList
              dataKey={'player'}
              position={'insideRight'}
              content={this.renderPlayerLabel}
            />
            <LabelList dataKey={'imageThumb'} position={'left'} content={this.renderImageLabels} />
            {data.map((entry: IVoteCount, i: number) => {
              return <Cell key={i} fill={theme.secondary} />;
            })}
          </Bar>
        </BarChart>
      </ResponsiveContainer>
    );
  }
}
