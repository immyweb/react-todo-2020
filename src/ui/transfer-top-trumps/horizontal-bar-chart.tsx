import * as React from 'react';
import * as DOM from 'react-dom';

import { fonts } from '../../styles/fonts';

import { ResponsiveContainer, BarChart, Bar, Cell, XAxis, YAxis, LabelList } from 'recharts';

import { BarWrapper, BarContainer } from './styles';
import { colours } from '../../styles/colours';

import { IVoteCount } from './types';
import { ITheme } from '../../types/theme';

interface IProps {
  data: IVoteCount[];
  theme: ITheme;
}

export class ChartHorizontalBar extends React.Component<IProps, { width: number }> {
  private chartElement: HTMLElement;

  constructor(props: IProps) {
    super(props);
    this.state = {
      width: 0,
    };
  }

  componentDidMount() {
    this.chartElement = DOM.findDOMNode(this) as HTMLElement;
    this.setState({ width: this.chartElement.offsetWidth });
  }

  renderLabels = (props: any) => {
    const { y, height, value } = props;

    return (
      <text
        x={this.state.width - 10}
        y={y + height / 2 + 1}
        fill="#fff"
        textAnchor="middle"
        style={{ fontFamily: fonts.$theSunMedium, fontSize: 10 }}
        dominantBaseline="middle"
      >
        {`${value}%`}
      </text>
    );
  };

  render() {
    const { data, theme } = this.props;
    return (
      <BarWrapper>
        <BarContainer>
          <ResponsiveContainer width="100%" height={380} minHeight={380}>
            <BarChart
              data={data}
              layout={'vertical'}
              barGap={3}
              margin={{ top: 8, right: 30, left: 60, bottom: 20 }}
            >
              <XAxis type={'number'} axisLine={false} hide={true} />
              <YAxis dataKey={'player'} type={'category'} hide={true} />
              <Bar dataKey={'votePercent'} barSize={14} background={false}>
                <LabelList
                  dataKey={'votePercent'}
                  position={'outside'}
                  content={this.renderLabels}
                />
                {data.map((entry: IVoteCount, i: number) => {
                  return <Cell key={i} fill={theme.primary} />;
                })}
              </Bar>
              <Bar dataKey={'noVotePercent'} barSize={14} background={false}>
                <LabelList
                  dataKey={'noVotePercent'}
                  position={'outside'}
                  content={this.renderLabels}
                />
                {data.map((entry: IVoteCount, i: number) => {
                  return <Cell key={i} fill={colours.$main} />;
                })}
              </Bar>
            </BarChart>
          </ResponsiveContainer>
        </BarContainer>
      </BarWrapper>
    );
  }
}
