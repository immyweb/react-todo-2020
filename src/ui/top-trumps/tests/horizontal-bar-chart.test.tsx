import * as React from 'react';
import { shallow } from 'enzyme';

import { ChartHorizontalBar } from '../horizontal-bar-chart';

const data = [
  {
    player: 'Cameron Small',
    voteCount: 50,
    noVoteCount: 0,
    votePercent: 80,
    imageThumb: 'handsome.jpg',
  },
  {
    player: 'Laura Kingston',
    voteCount: 10,
    noVoteCount: 0,
    votePercent: 10,
    imageThumb: 'hot.jpg',
  },
  {
    player: 'Maramead',
    voteCount: 11,
    noVoteCount: 0,
    votePercent: 10,
    imageThumb: 'ironman.gif',
  },
];

const theme = {
  primary: '#ffffff',
  secondary: '#000000',
};

function shallowRender(props?: any) {
  return shallow(<ChartHorizontalBar data={data} theme={theme} {...props} />, {
    disableLifecycleMethods: true,
  });
}

describe('<ChartHorizontalBar />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });
});
