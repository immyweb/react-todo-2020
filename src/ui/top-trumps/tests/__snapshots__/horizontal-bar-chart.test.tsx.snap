// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartHorizontalBar /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChartHorizontalBar
    data={
      Array [
        Object {
          "imageThumb": "handsome.jpg",
          "noVoteCount": 0,
          "player": "Cameron Small",
          "voteCount": 50,
          "votePercent": 80,
        },
        Object {
          "imageThumb": "hot.jpg",
          "noVoteCount": 0,
          "player": "Laura Kingston",
          "voteCount": 10,
          "votePercent": 10,
        },
        Object {
          "imageThumb": "ironman.gif",
          "noVoteCount": 0,
          "player": "Maramead",
          "voteCount": 11,
          "votePercent": 10,
        },
      ]
    }
    theme={
      Object {
        "primary": "#ffffff",
        "secondary": "#000000",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <BarChart
        barCategoryGap="10%"
        barGap={10}
        data={
          Array [
            Object {
              "imageThumb": "handsome.jpg",
              "noVoteCount": 0,
              "player": "Cameron Small",
              "voteCount": 50,
              "votePercent": 80,
            },
            Object {
              "imageThumb": "hot.jpg",
              "noVoteCount": 0,
              "player": "Laura Kingston",
              "voteCount": 10,
              "votePercent": 10,
            },
            Object {
              "imageThumb": "ironman.gif",
              "noVoteCount": 0,
              "player": "Maramead",
              "voteCount": 11,
              "votePercent": 10,
            },
          ]
        }
        layout="vertical"
        margin={
          Object {
            "left": -5,
          }
        }
        reverseStackOrder={false}
        stackOffset="none"
      >
        <XAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          axisLine={false}
          domain={
            Array [
              0,
              100,
            ]
          }
          height={30}
          hide={true}
          mirror={false}
          orientation="bottom"
          padding={
            Object {
              "left": 0,
              "right": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="number"
          width={0}
          xAxisId={0}
        />
        <YAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          dataKey="player"
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={0}
          hide={true}
          mirror={false}
          orientation="left"
          padding={
            Object {
              "bottom": 0,
              "top": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="category"
          width={60}
          yAxisId={0}
        />
        <Bar
          animationBegin={0}
          animationDuration={400}
          animationEasing="ease"
          background={
            Object {
              "fill": "#adafb0",
              "fillOpacity": 0.4,
            }
          }
          barSize={-10}
          data={Array []}
          dataKey="votePercentage"
          hide={false}
          isAnimationActive={true}
          layout="vertical"
          legendType="rect"
          minPointSize={0}
          onAnimationEnd={[Function]}
          onAnimationStart={[Function]}
          xAxisId={0}
          yAxisId={0}
        >
          <LabelList
            content={[Function]}
            dataKey="votePercentage"
            position="insideRight"
            valueAccessor={[Function]}
          />
          <LabelList
            content={[Function]}
            dataKey="player"
            position="insideRight"
            valueAccessor={[Function]}
          />
          <LabelList
            content={[Function]}
            dataKey="imageThumb"
            position="left"
            valueAccessor={[Function]}
          />
          <Cell
            fill="#000000"
          />
          <Cell
            fill="#000000"
          />
          <Cell
            fill="#000000"
          />
        </Bar>
      </BarChart>,
      "debounce": 0,
      "height": "100%",
      "width": "100%",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "barCategoryGap": "10%",
        "barGap": 10,
        "children": Array [
          <XAxis
            allowDataOverflow={false}
            allowDecimals={true}
            allowDuplicatedCategory={true}
            axisLine={false}
            domain={
              Array [
                0,
                100,
              ]
            }
            height={30}
            hide={true}
            mirror={false}
            orientation="bottom"
            padding={
              Object {
                "left": 0,
                "right": 0,
              }
            }
            reversed={false}
            scale="auto"
            tickCount={5}
            type="number"
            width={0}
            xAxisId={0}
          />,
          <YAxis
            allowDataOverflow={false}
            allowDecimals={true}
            allowDuplicatedCategory={true}
            dataKey="player"
            domain={
              Array [
                0,
                "auto",
              ]
            }
            height={0}
            hide={true}
            mirror={false}
            orientation="left"
            padding={
              Object {
                "bottom": 0,
                "top": 0,
              }
            }
            reversed={false}
            scale="auto"
            tickCount={5}
            type="category"
            width={60}
            yAxisId={0}
          />,
          <Bar
            animationBegin={0}
            animationDuration={400}
            animationEasing="ease"
            background={
              Object {
                "fill": "#adafb0",
                "fillOpacity": 0.4,
              }
            }
            barSize={-10}
            data={Array []}
            dataKey="votePercentage"
            hide={false}
            isAnimationActive={true}
            layout="vertical"
            legendType="rect"
            minPointSize={0}
            onAnimationEnd={[Function]}
            onAnimationStart={[Function]}
            xAxisId={0}
            yAxisId={0}
          >
            <LabelList
              content={[Function]}
              dataKey="votePercentage"
              position="insideRight"
              valueAccessor={[Function]}
            />
            <LabelList
              content={[Function]}
              dataKey="player"
              position="insideRight"
              valueAccessor={[Function]}
            />
            <LabelList
              content={[Function]}
              dataKey="imageThumb"
              position="left"
              valueAccessor={[Function]}
            />
            <Cell
              fill="#000000"
            />
            <Cell
              fill="#000000"
            />
            <Cell
              fill="#000000"
            />
          </Bar>,
        ],
        "data": Array [
          Object {
            "imageThumb": "handsome.jpg",
            "noVoteCount": 0,
            "player": "Cameron Small",
            "voteCount": 50,
            "votePercent": 80,
          },
          Object {
            "imageThumb": "hot.jpg",
            "noVoteCount": 0,
            "player": "Laura Kingston",
            "voteCount": 10,
            "votePercent": 10,
          },
          Object {
            "imageThumb": "ironman.gif",
            "noVoteCount": 0,
            "player": "Maramead",
            "voteCount": 11,
            "votePercent": 10,
          },
        ],
        "layout": "vertical",
        "margin": Object {
          "left": -5,
        },
        "reverseStackOrder": false,
        "stackOffset": "none",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowDataOverflow": false,
            "allowDecimals": true,
            "allowDuplicatedCategory": true,
            "axisLine": false,
            "domain": Array [
              0,
              100,
            ],
            "height": 30,
            "hide": true,
            "mirror": false,
            "orientation": "bottom",
            "padding": Object {
              "left": 0,
              "right": 0,
            },
            "reversed": false,
            "scale": "auto",
            "tickCount": 5,
            "type": "number",
            "width": 0,
            "xAxisId": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowDataOverflow": false,
            "allowDecimals": true,
            "allowDuplicatedCategory": true,
            "dataKey": "player",
            "domain": Array [
              0,
              "auto",
            ],
            "height": 0,
            "hide": true,
            "mirror": false,
            "orientation": "left",
            "padding": Object {
              "bottom": 0,
              "top": 0,
            },
            "reversed": false,
            "scale": "auto",
            "tickCount": 5,
            "type": "category",
            "width": 60,
            "yAxisId": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animationBegin": 0,
            "animationDuration": 400,
            "animationEasing": "ease",
            "background": Object {
              "fill": "#adafb0",
              "fillOpacity": 0.4,
            },
            "barSize": -10,
            "children": Array [
              <LabelList
                content={[Function]}
                dataKey="votePercentage"
                position="insideRight"
                valueAccessor={[Function]}
              />,
              <LabelList
                content={[Function]}
                dataKey="player"
                position="insideRight"
                valueAccessor={[Function]}
              />,
              <LabelList
                content={[Function]}
                dataKey="imageThumb"
                position="left"
                valueAccessor={[Function]}
              />,
              Array [
                <Cell
                  fill="#000000"
                />,
                <Cell
                  fill="#000000"
                />,
                <Cell
                  fill="#000000"
                />,
              ],
            ],
            "data": Array [],
            "dataKey": "votePercentage",
            "hide": false,
            "isAnimationActive": true,
            "layout": "vertical",
            "legendType": "rect",
            "minPointSize": 0,
            "onAnimationEnd": [Function],
            "onAnimationStart": [Function],
            "xAxisId": 0,
            "yAxisId": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "content": [Function],
                "dataKey": "votePercentage",
                "position": "insideRight",
                "valueAccessor": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "content": [Function],
                "dataKey": "player",
                "position": "insideRight",
                "valueAccessor": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "content": [Function],
                "dataKey": "imageThumb",
                "position": "left",
                "valueAccessor": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "fill": "#000000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "fill": "#000000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "fill": "#000000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <BarChart
          barCategoryGap="10%"
          barGap={10}
          data={
            Array [
              Object {
                "imageThumb": "handsome.jpg",
                "noVoteCount": 0,
                "player": "Cameron Small",
                "voteCount": 50,
                "votePercent": 80,
              },
              Object {
                "imageThumb": "hot.jpg",
                "noVoteCount": 0,
                "player": "Laura Kingston",
                "voteCount": 10,
                "votePercent": 10,
              },
              Object {
                "imageThumb": "ironman.gif",
                "noVoteCount": 0,
                "player": "Maramead",
                "voteCount": 11,
                "votePercent": 10,
              },
            ]
          }
          layout="vertical"
          margin={
            Object {
              "left": -5,
            }
          }
          reverseStackOrder={false}
          stackOffset="none"
        >
          <XAxis
            allowDataOverflow={false}
            allowDecimals={true}
            allowDuplicatedCategory={true}
            axisLine={false}
            domain={
              Array [
                0,
                100,
              ]
            }
            height={30}
            hide={true}
            mirror={false}
            orientation="bottom"
            padding={
              Object {
                "left": 0,
                "right": 0,
              }
            }
            reversed={false}
            scale="auto"
            tickCount={5}
            type="number"
            width={0}
            xAxisId={0}
          />
          <YAxis
            allowDataOverflow={false}
            allowDecimals={true}
            allowDuplicatedCategory={true}
            dataKey="player"
            domain={
              Array [
                0,
                "auto",
              ]
            }
            height={0}
            hide={true}
            mirror={false}
            orientation="left"
            padding={
              Object {
                "bottom": 0,
                "top": 0,
              }
            }
            reversed={false}
            scale="auto"
            tickCount={5}
            type="category"
            width={60}
            yAxisId={0}
          />
          <Bar
            animationBegin={0}
            animationDuration={400}
            animationEasing="ease"
            background={
              Object {
                "fill": "#adafb0",
                "fillOpacity": 0.4,
              }
            }
            barSize={-10}
            data={Array []}
            dataKey="votePercentage"
            hide={false}
            isAnimationActive={true}
            layout="vertical"
            legendType="rect"
            minPointSize={0}
            onAnimationEnd={[Function]}
            onAnimationStart={[Function]}
            xAxisId={0}
            yAxisId={0}
          >
            <LabelList
              content={[Function]}
              dataKey="votePercentage"
              position="insideRight"
              valueAccessor={[Function]}
            />
            <LabelList
              content={[Function]}
              dataKey="player"
              position="insideRight"
              valueAccessor={[Function]}
            />
            <LabelList
              content={[Function]}
              dataKey="imageThumb"
              position="left"
              valueAccessor={[Function]}
            />
            <Cell
              fill="#000000"
            />
            <Cell
              fill="#000000"
            />
            <Cell
              fill="#000000"
            />
          </Bar>
        </BarChart>,
        "debounce": 0,
        "height": "100%",
        "width": "100%",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "barCategoryGap": "10%",
          "barGap": 10,
          "children": Array [
            <XAxis
              allowDataOverflow={false}
              allowDecimals={true}
              allowDuplicatedCategory={true}
              axisLine={false}
              domain={
                Array [
                  0,
                  100,
                ]
              }
              height={30}
              hide={true}
              mirror={false}
              orientation="bottom"
              padding={
                Object {
                  "left": 0,
                  "right": 0,
                }
              }
              reversed={false}
              scale="auto"
              tickCount={5}
              type="number"
              width={0}
              xAxisId={0}
            />,
            <YAxis
              allowDataOverflow={false}
              allowDecimals={true}
              allowDuplicatedCategory={true}
              dataKey="player"
              domain={
                Array [
                  0,
                  "auto",
                ]
              }
              height={0}
              hide={true}
              mirror={false}
              orientation="left"
              padding={
                Object {
                  "bottom": 0,
                  "top": 0,
                }
              }
              reversed={false}
              scale="auto"
              tickCount={5}
              type="category"
              width={60}
              yAxisId={0}
            />,
            <Bar
              animationBegin={0}
              animationDuration={400}
              animationEasing="ease"
              background={
                Object {
                  "fill": "#adafb0",
                  "fillOpacity": 0.4,
                }
              }
              barSize={-10}
              data={Array []}
              dataKey="votePercentage"
              hide={false}
              isAnimationActive={true}
              layout="vertical"
              legendType="rect"
              minPointSize={0}
              onAnimationEnd={[Function]}
              onAnimationStart={[Function]}
              xAxisId={0}
              yAxisId={0}
            >
              <LabelList
                content={[Function]}
                dataKey="votePercentage"
                position="insideRight"
                valueAccessor={[Function]}
              />
              <LabelList
                content={[Function]}
                dataKey="player"
                position="insideRight"
                valueAccessor={[Function]}
              />
              <LabelList
                content={[Function]}
                dataKey="imageThumb"
                position="left"
                valueAccessor={[Function]}
              />
              <Cell
                fill="#000000"
              />
              <Cell
                fill="#000000"
              />
              <Cell
                fill="#000000"
              />
            </Bar>,
          ],
          "data": Array [
            Object {
              "imageThumb": "handsome.jpg",
              "noVoteCount": 0,
              "player": "Cameron Small",
              "voteCount": 50,
              "votePercent": 80,
            },
            Object {
              "imageThumb": "hot.jpg",
              "noVoteCount": 0,
              "player": "Laura Kingston",
              "voteCount": 10,
              "votePercent": 10,
            },
            Object {
              "imageThumb": "ironman.gif",
              "noVoteCount": 0,
              "player": "Maramead",
              "voteCount": 11,
              "votePercent": 10,
            },
          ],
          "layout": "vertical",
          "margin": Object {
            "left": -5,
          },
          "reverseStackOrder": false,
          "stackOffset": "none",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowDataOverflow": false,
              "allowDecimals": true,
              "allowDuplicatedCategory": true,
              "axisLine": false,
              "domain": Array [
                0,
                100,
              ],
              "height": 30,
              "hide": true,
              "mirror": false,
              "orientation": "bottom",
              "padding": Object {
                "left": 0,
                "right": 0,
              },
              "reversed": false,
              "scale": "auto",
              "tickCount": 5,
              "type": "number",
              "width": 0,
              "xAxisId": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowDataOverflow": false,
              "allowDecimals": true,
              "allowDuplicatedCategory": true,
              "dataKey": "player",
              "domain": Array [
                0,
                "auto",
              ],
              "height": 0,
              "hide": true,
              "mirror": false,
              "orientation": "left",
              "padding": Object {
                "bottom": 0,
                "top": 0,
              },
              "reversed": false,
              "scale": "auto",
              "tickCount": 5,
              "type": "category",
              "width": 60,
              "yAxisId": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "animationBegin": 0,
              "animationDuration": 400,
              "animationEasing": "ease",
              "background": Object {
                "fill": "#adafb0",
                "fillOpacity": 0.4,
              },
              "barSize": -10,
              "children": Array [
                <LabelList
                  content={[Function]}
                  dataKey="votePercentage"
                  position="insideRight"
                  valueAccessor={[Function]}
                />,
                <LabelList
                  content={[Function]}
                  dataKey="player"
                  position="insideRight"
                  valueAccessor={[Function]}
                />,
                <LabelList
                  content={[Function]}
                  dataKey="imageThumb"
                  position="left"
                  valueAccessor={[Function]}
                />,
                Array [
                  <Cell
                    fill="#000000"
                  />,
                  <Cell
                    fill="#000000"
                  />,
                  <Cell
                    fill="#000000"
                  />,
                ],
              ],
              "data": Array [],
              "dataKey": "votePercentage",
              "hide": false,
              "isAnimationActive": true,
              "layout": "vertical",
              "legendType": "rect",
              "minPointSize": 0,
              "onAnimationEnd": [Function],
              "onAnimationStart": [Function],
              "xAxisId": 0,
              "yAxisId": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "content": [Function],
                  "dataKey": "votePercentage",
                  "position": "insideRight",
                  "valueAccessor": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "content": [Function],
                  "dataKey": "player",
                  "position": "insideRight",
                  "valueAccessor": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "content": [Function],
                  "dataKey": "imageThumb",
                  "position": "left",
                  "valueAccessor": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "function",
                "props": Object {
                  "fill": "#000000",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {
                  "fill": "#000000",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "function",
                "props": Object {
                  "fill": "#000000",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
